app/src/main/java/ru/vssemikoz/newsfeed/MainActivity.java

```diff
@@ -24,46 +26,60 @@
 public class MainActivity extends AppCompatActivity {
     String KEY = "c94a57cbbb50497f94a2bb167dc91fc5";
     List<NewsItem> newsItems = new ArrayList<NewsItem>();
+    NewsAppDataBase newsDataBase;
+    NewsItemDAO newsItemDAO;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
+        initDataBase();
         Callback<NewsItemList> callbackNewsItemList = new Callback<NewsItemList>() {
             @Override
             public void onResponse(Call<NewsItemList> call, Response<NewsItemList> response) {
                 if (!response.isSuccessful()){
                     Log.d("MyLog", "onResponse " + response.code());
                     return;
                 }
-                newsItems = Objects.requireNonNull(response.body()).getNewsItem();
-                Log.d("MyLog", "onSuccess " + newsItems);
+                newsItemDAO.insertUnique(response.body().getNewsItem());
```

Comment:
Please, take a look at the comment for the previous PR regarding the sets of models. That fix will probably affect this piece of code.